### ISSUE TITLE
apr finetune: Add --task instruct for generative instruction-following LoRA training

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#371]
## Problem

`apr finetune --task classify` works end-to-end for classification (SafetySample corpus → ClassifyPipeline → ClassifyTrainer → checkpoints). But there is no `--task instruct` path for generative instruction-following fine-tuning.

The `TestGenCorpus` in `entrenar/src/finetune/corpus.rs` has `format_prompt()` and `format_target()` using Qwen's `<|im_start|>` chat template, but this is dead code — never wired to the CLI.

## Impact

The apr-leaderboard project needs `apr finetune --task instruct` to fine-tune Qwen2.5-Coder-7B-Instruct on custom instruction data for BigCodeBench submission. Without this, the entire pipeline from data prep → LoRA finetune → eval is blocked.

## Required Components

Following the `--task classify` pattern in `finetune.rs`:

1. **InstructSample** corpus loader (`{"instruction": "...", "response": "..."}` JSONL)
2. **InstructPipeline** — transformer + causal LM head (next-token prediction)
3. **InstructTrainer** — epoch loop with causal LM loss, validation perplexity, checkpointing
4. **CLI wiring** — `--task instruct` dispatch in `run()` at line 375

### Contracts

- F-INST-001: Token sequence = tokenize(format_prompt(instruction)) + tokenize(response) + EOS
- F-INST-002: Loss computed only on response tokens (not prompt tokens)
- F-INST-003: Perplexity reported per-epoch on validation split
- F-INST-004: LoRA adapters saved in APR format (same as classify path)

### JSONL Format

```json
{"instruction": "Write a function using pandas to filter rows", "response": "import pandas as pd\n\ndef filter_rows(df, col, threshold):\n    return df[df[col] > threshold]"}
```

## Existing Infrastructure to Reuse

- `TestGenCorpus::format_prompt()` — Qwen `<|im_start|>` chat template
- `ClassifyTrainer` — epoch management, validation split, early stopping, LR scheduling
- `ClassifyPipeline` — model loading, LoRA adapter creation
- `cross_entropy_loss()` — per-token loss (extend to sequence)

## References

- Classification pipeline: `entrenar/src/finetune/classify_pipeline.rs`
- TestGenCorpus: `entrenar/src/finetune/corpus.rs:231-257`
- CLI dispatch: `aprender/crates/apr-cli/src/commands/finetune.rs:374-390`
